NExtends/Expressions/ParameterChanger.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Linq.Expressions;
+
+namespace NExtends.Expressions
+{
+	public class ParameterChanger : ExpressionVisitor
+	{
+		ParameterExpression RemplacingParameter { get; set; }
+		ParameterExpression RemplacedParameter { get; set; }
+
+		ParameterChanger(ParameterExpression remplacedParameter, ParameterExpression remplacingParameter)
+		{
+			RemplacedParameter = remplacedParameter;
+			RemplacingParameter = remplacingParameter;
+		}
+
+		public static Expression<Func<T, bool>> ChangeParameter<T>(Expression<Func<T, bool>> lambda, ParameterExpression replacingParameter)
+		{
+			var newBody = GetChangedParameterBody(lambda, replacingParameter);
+			return Expression.Lambda<Func<T, bool>>(newBody, replacingParameter);
+		}
+
+		public static Expression GetChangedParameterBody<T>(Expression<Func<T, bool>> lambda, ParameterExpression replacingParameter)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Linq.Expressions.LambdaExpression' instead of 'System.Linq.Expressions.Expression<System.Func<T, bool>>'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)